cmd/prometheus/config.go

```diff
@@ -52,15 +51,37 @@ var cfg = struct {
 	notifierTimeout    time.Duration
 	queryEngine        promql.EngineOptions
 	web                web.Options
-	remote             remote.Options
 
 	alertmanagerURLs stringset
 	prometheusURL    string
-	influxdbURL      string
 }{
 	alertmanagerURLs: stringset{},
 }
 
+// Value type for flags that are now unused, but which are kept around to
+// fulfill 1.0 stability guarantees.
+type unusedFlag struct {
+	name  string
+	value string
+	help  string
+}
+
+func (f *unusedFlag) Set(v string) error {
+	f.value = v
+	log.Errorf("ERROR: flag %q is unused, but set to %q! See the flag's help message: %s", f.name, f.value, f.help)
```

Comment:
For me warnings are things like "this is depreacted, but I updated it for you. It might not work in the future" or "I don't know what you mean here, but I'll just ignore it and continue". Errors are "I tried to do something and I failed". So I'd still go with warning. I don't feel strongly about it though.